### PR TITLE
MRG: Fix displayed Raw duration in Jupyter notebook

### LINKED ISSUE #11202: Incorrect raw duration displayed in HTML repr
_Originally posted by @hoechenberger in https://github.com/mne-tools/mne-python/issues/11196#issuecomment-1257090412_

btw there's a bug in the HTML repr that is not present in the text repr:

Screen Shot 2022-09-25 at 02 29 36

```python
import numpy as np
import mne

n_seconds = 1
sfreq = 100
ch_names = ['1']
data = np.ones((len(ch_names), sfreq*n_seconds))
info = mne.create_info(ch_names=ch_names, sfreq=sfreq)
raw = mne.io.RawArray(data, info)
```

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes #11202

```python
# %%
import numpy as np
import mne

n_seconds = 1
sfreq = 100
ch_names = ['1']
data = np.ones((len(ch_names), sfreq*n_seconds))
info = mne.create_info(ch_names=ch_names, sfreq=sfreq)
raw = mne.io.RawArray(data, info)
raw
```

`main`:
Screen Shot 2022-09-25 at 13 43 28

PR:
Screen Shot 2022-09-25 at 13 43 06

I'm don't feel like adding a regression test for this…

### CODE REVIEW

File: mne/io/base.py [hunk 1]
```diff
@@ -1745,9 +1745,15 @@ def _repr_html_(self, caption=None):
         basenames = [
             os.path.basename(f) for f in self._filenames if f is not None
         ]
-        m, s = divmod(self._last_time - self.first_time, 60)
-        h, m = divmod(m, 60)
-        duration = f'{int(h):02d}:{int(m):02d}:{int(s):02d}'
+
+        # https://stackoverflow.com/a/10981895
+        duration = timedelta(seconds=self._last_time - self.first_time)
```
Comment (hoechenberger) [inline mne/io/base.py:1750]: We use a different calculation than in `__repr__()` here. In `__repr__()`, we use `self.times[-1]`, which I believe is a bug.

cc @larsoner @agramfort

Comment (hoechenberger) [reply mne/io/base.py:1750]: Never mind, it appears `raw.times` is already adjusted for `first_time`.

File: mne/io/base.py [hunk 2]
```diff
@@ -1745,9 +1745,15 @@ def _repr_html_(self, caption=None):
         basenames = [
             os.path.basename(f) for f in self._filenames if f is not None
         ]
-        m, s = divmod(self._last_time - self.first_time, 60)
-        h, m = divmod(m, 60)
-        duration = f'{int(h):02d}:{int(m):02d}:{int(s):02d}'
+
+        # https://stackoverflow.com/a/10981895
+        duration = timedelta(seconds=raw.times[-1])
```
Comment (hoechenberger) [inline mne/io/base.py:1750]:
```suggestion
        duration = timedelta(seconds=self.times[-1])
```

**Review by agramfort — COMMENTED** [review]
Can factorize the code between repr and repr_html in a private function? Also add a test ? Did you consider using strftime to format the time ?

**Comment by hoechenberger** [comment]
I only wanted to provide a quick and dirty fix here. Repr displays time in seconds, weil in HTML we show it in HH:MM:SS so this is not directly exchangeable anyway ... any suggestions now to unify this?

**Comment by hoechenberger** [comment]
CI errors are unrelated

**Comment by larsoner** [comment]
I think it's safe enough to unify separately, thanks for the quick fix @hoechenberger !

**Pull request merged**